regex support non-whitespace language

w7lau opened the pull request:
add non-withespace language list

**Proposed changes**:
- add the language pass to pattern_utils
- check the language if non-withespace so do not add the word boundary

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)

non-withespace language like Chinese should not add the word boundary

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RasaHQ/rasa?pullRequest=7107) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/RasaHQ/rasa?pullRequest=7107) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RasaHQ/rasa?pullRequest=7107) it.</sub>

sara-tagger commented:
Thanks for submitting a pull request 🚀 @erohmensing will take a look at it as soon as possible ✨

erohmensing commented on rasa/nlu/utils/pattern_utils.py:
```diff
@@ -49,7 +53,9 @@ def _generate_lookup_regex(lookup_table: Dict[Text, Union[Text, List[Text]]]) ->
     elements_sanitized = [re.escape(e) for e in elements_to_regex]
-
+    not_boundary_language = ["zh", "ja", "th"]
+    if str(config_language).lower in not_boundary_language:
```
```suggestion
    if str(config_language).lower() in not_boundary_language:
```

erohmensing commented on rasa/nlu/utils/pattern_utils.py:
```diff
@@ -49,7 +53,9 @@ def _generate_lookup_regex(lookup_table: Dict[Text, Union[Text, List[Text]]]) ->
     # sanitize the regex, escape special characters
     elements_sanitized = [re.escape(e) for e in elements_to_regex]
-
+    not_boundary_language = ["zh", "ja", "th"]
```
can we move these into a constant `NON_WHITESPACE_LANGUAGES` in `rasa/nlu/constants.py`?

w7lau replied:
Of course !

erohmensing commented on rasa/nlu/utils/pattern_utils.py:
```diff
@@ -21,13 +22,16 @@ def _convert_lookup_tables_to_regex(
-def _generate_lookup_regex(lookup_table: Dict[Text, Union[Text, List[Text]]]) -> Text:
+def _generate_lookup_regex(
+        lookup_table: Dict[Text, Union[Text, List[Text]]],
+        config_language: Text = None,
```
Each config must have a language, I can't think of a time where this would be `None`, can you? If we provide `None` for this, we'll get an exception when we try to `lower()` it.

w7lau replied:
I am a newer in python, so i don't know what will happen and how it work, i will listen to your suggestions

erohmensing reviewed (changes requested):
Thanks for the PR! This improvement makes a lot of sense 👌 I have some suggestions around the implementation. 

Would you also do a few things as per the PR template:
- format the code with `black`, otherwise it will not pass CI
- add a changelog entry, so that other users using non-whitespaced tokenized languages know they can use this feature 
- add a test in the `test_patten_utils.py`

erohmensing commented:
Going to remove my request until you're ready for another review after addressing comments 🙂

w7lau commented:
> Going to remove my request until you're ready for another review after addressing comments 🙂



> Going to remove my request until you're ready for another review after addressing comments 🙂

It's ok, i change the .lower() and check the language is not none, if none let it go to default code.

w7lau commented:
> Thanks for the PR! This improvement makes a lot of sense 👌 I have some suggestions around the implementation.
> 
> Would you also do a few things as per the PR template:
> 
> * format the code with `black`, otherwise it will not pass CI
> * add a changelog entry, so that other users using non-whitespaced tokenized languages know they can use this feature
> * add a test in the `test_patten_utils.py`

Ok, i will do this.

w7lau commented:
@erohmensing What issue type should i use in changelog?

w7lau commented:
> @erohmensing What issue type should i use in changelog?

And how can i know the test_patten_utils.py is successful ?
